packages/static-config/src/swc.ts

```diff
@@ -0,0 +1,188 @@
+import type {
+  ArrayExpression,
+  BooleanLiteral,
+  ExportDeclaration,
+  Identifier,
+  KeyValueProperty,
+  Module,
+  Node,
+  NullLiteral,
+  NumericLiteral,
+  ObjectExpression,
+  StringLiteral,
+  VariableDeclaration,
+} from '@swc/core';
+import { FromSchema, JSONSchema } from 'json-schema-to-ts';
+import { validate } from './validation';
+
+function isExportDeclaration(node: Node): node is ExportDeclaration {
+  return node.type === 'ExportDeclaration';
+}
+
+function isVariableDeclaration(node: Node): node is VariableDeclaration {
+  return node.type === 'VariableDeclaration';
+}
+
+function isIdentifier(node: Node): node is Identifier {
+  return node.type === 'Identifier';
+}
+
+function isBooleanLiteral(node: Node): node is BooleanLiteral {
+  return node.type === 'BooleanLiteral';
+}
+
+function isNullLiteral(node: Node): node is NullLiteral {
+  return node.type === 'NullLiteral';
+}
+
+function isStringLiteral(node: Node): node is StringLiteral {
+  return node.type === 'StringLiteral';
+}
+
+function isNumericLiteral(node: Node): node is NumericLiteral {
+  return node.type === 'NumericLiteral';
+}
+
+function isArrayExpression(node: Node): node is ArrayExpression {
+  return node.type === 'ArrayExpression';
+}
+
+function isObjectExpression(node: Node): node is ObjectExpression {
+  return node.type === 'ObjectExpression';
+}
+
+function isKeyValueProperty(node: Node): node is KeyValueProperty {
+  return node.type === 'KeyValueProperty';
+}
+
+export type Value = undefined | null | boolean | string | number | any[] | Record<string, any>;
+export class UnsupportedValueError extends Error {}
+
+function extractValue(node: Node): Value {
```

Comment:
```suggestion
function extractValue(node: Expression): Value {
```

This should remove the need for all the additional functions. You can just compare `node.type === 'FOO'` to narrow inline

Reply:
Aha! Thanks for the help 😄